localstack-core/localstack/services/lambda_/provider.py

```diff
@@ -1892,11 +1892,17 @@ def check_service_resource_exists(
         )
         if service in ["sqs", "sqs-fifo"]:
             try:
-                source_client.get_queue_attributes(QueueUrl=arn["resource"])
+                # AWS uses `GetQueueAttributes` internally to verify the queue existence, but we need the `QueueUrl`
+                # which is not given directly. We build out a dummy `QueueUrl` which can be parsed by SQS to return
+                # the right value
+                queue_name = arn["resource"].split("/")[-1]
+                queue_url = f"http://sqs.{arn['region']}.domain/{arn['account']}/{queue_name}"
```

Comment:
Interesting hack. I wonder how AWS does it internally because [GetQueueUrl](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_GetQueueUrl.html) is not a required IAM permission: https://repost.aws/knowledge-center/lambda-sqs-event-source

Reply:
Yes, I really wonder too.. maybe just rebuilding the queue manually like us, but it seems… weird 😅 I don’t know if there would be any other operation that sqs:getQueueAttributes permission would give access to. Thanks for the review!